질문하기 위해 브런치 Header 생성

94CJG opened the pull request:
ㅁ 질문 : 스타일 컴포넌트를 사용하여 변화를 주고 싶은데 어떻게 해야할까?

ㅁ 각각 스타일 컴포넌트를 만들어서 return 부분에 넣어주면 될거 같지만,
코드가 가독성이 안좋아 질 거 같은데 효율적인 다른 방법이 없을까?

ㅁ css 파일을 만들거나 or 직접적으로 html css를 코드에 작성하면 편하기는 하겠지만 지저분 해질 것 같아서 사용을 안하려고 한다.

94CJG committed:
질문하기 위해 브런치 Header 생성

ㅁ 질문 : 스타일 컴포넌트를 사용하여 변화를 주고 싶은데 어떻게 해야할까?

ㅁ 각각 스타일 컴포넌트를 만들어서 return 부분에 넣어주면 될거 같지만,
코드가 가독성이 안좋아 질 거 같은데 효율적인 다른 방법이 없을까?

ㅁ css 파일을 만들거나 or 직접적으로 html css를 코드에 작성하면 편하기는 하겠지만 지저분 해질 것 같아서 사용을 안하려고 한다.

Kwakcena commented:
코드로 먼저 시도해보면 좋을 것 같아.
어떤 시도를 했는지 내가 봐야 질문을 이해할 수 있거든.
> 각각 스타일 컴포넌트를 만들어서 return 부분에 넣어주면 될거 같지만, 코드가 가독성이 안좋아 질 거 같은데 효율적인 다른 방법이 없을까?

라고 했는데, 여기서 return 부분은 어디이고, 코드 가독성은 왜 안좋아지는지 해보지 않았기 때문에 모르는 일이야.
한번 해 보고나서 `가독성이 xx 해서 안좋아지는것 같다` 라고 질문을 남겨줘. 그럼 그거에 리뷰를 달아줄게.

94CJG committed:
Header 컴포넌트 및 스타일 컴포넌트를 이용하여 마크업

ㅁ 스타일 컴포넌트를 사용하여 마크업
ㅁ

94CJG committed:
Header 컴포넌트 및 스타일 컴포넌트를 통해 마크업 함.

ㅁ 구조만 이렇게 작성을 해봤는데 더 좋은 방법이 있는지?
ㅁ 스타일 컴포넌트를 현재 코드처럼 이런식으로 사용 하는게 맞는지? 아니라면 더 좋은 방법이 뭐가 있을지?
ㅁ 지금 상태로 스타일 컴포넌트로 적용 시켜서 코드를 작성하게 되면 가독성 괜찮은지? (현재 마크업을 뜻함)

94CJG committed:
main, HotItems 컴포넌드 생성

ㅁ 브런치를 새로 만들어서 했어야 했는데 익숙하지 않아서 header 브런치에 만듦.

ㅁ 문제: main에 있는 인기매물 버튼을 클릭시 HotItems.tsx 컴포넌트로 이동 하지를 않음.

ㅁ URL은 변경이 됐지만, 해당 컴포넌트 출력이 안됨.

ㅁ Router와 Switch, Route를 사용중

ㅁ 형조에게 질문 하기 위해서 push 진행

Kwakcena commented on src/App.tsx:
```diff
@@ -58,16 +61,21 @@ body
-      <GlobalStyle />
-      <Header />
-    </>
+    <BrowserRouter>
```
`<BrowserRouter>` 가 이미 `src/index.tsx` 파일에서 감싸고 있어서 여기서는 필요 없을 거 같아

Kwakcena commented on src/App.tsx:
```diff
@@ -58,16 +61,21 @@ body
+        <GlobalStyle />
+        <Header />
+        <Switch>
+          <Route path="">
```
`/` 경로로 유일하게 `<Main>` 컴포넌트만 보여주고 싶다면 이렇게 써야 해
```
<Route exact path="/">
```

94CJG replied:
exact를 코드를 입력 해주니까 됐어!!!!!!!!!!!!!!!! 감사합니다람쥐

Kwakcena commented on src/App.tsx:
```diff
@@ -58,16 +61,21 @@ body
+          <Route path="">
+            <Main />
+          </Route>
+          <Route path="/HotItems">
```
url 은 보통 [파스칼케이스](https://velog.io/@leyuri/%ED%91%9C%EA%B8%B0%EB%B2%95-%EC%8A%A4%EB%84%A4%EC%9D%B4%ED%81%AC-%EC%BC%80%EC%9D%B4%EC%8A%A4-%ED%8C%8C%EC%8A%A4%EC%B9%BC-%EC%BC%80%EC%9D%B4%EC%8A%A4-%EC%B9%B4%EB%A9%9C-%EC%BC%80%EC%9D%B4%EC%8A%A4) 를 쓰지 않아.

일반 소문자로 쓰는 게 좋을 듯
